online-grocery-store/src/pages/product/[product_id].js

```diff
@@ -0,0 +1,164 @@
+import { useState, useEffect } from "react";
+import Head from "next/head";
+import { useRouter } from "next/router";
+import {
+  Box,
+  Grid,
+  Image,
+  Text,
+  NumberFormatter,
+  Button,
+  Flex,
+} from "@mantine/core";
+import { useCart } from "../../store/store";
+import { Rating } from "@mantine/core";
+import Link from "next/link";
+
+const ProductPage = () => {
+  const addProduct = useCart((state) => state.addProduct);
+  const cart = useCart((state) => state.cart);
+  const [quantity, setQuantity] = useState(0);
+  console.log(cart);
+  const [product, setProduct] = useState();
+  const router = useRouter();
+  const { product_id } = router.query;
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(
+          `http://localhost:3000/api/getProduct?product_id=${product_id}`,
+        );
+        const product = await response.json();
+        setProduct(product);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchData();
+  }, [router.query]);
+  const decreaseQuantity = () => {
+    if (quantity > 0) {
+      setQuantity(quantity - 1);
```

Comment:
Read [this](https://legacy.reactjs.org/docs/hooks-reference.html#functional-updates)
`setQuantity(prevQuantity => prevQuantity - 1)`

Reply:
Done